### DISCUSSION TITLE
No available formula with the name "git".

### DISCUSSION THREAD

**Discussion by m1sfumo** [Homebrew/discussions#580] [Getting started] [answered]
i just installed homebrew with some errors leave.
i tried `brew update` and the errors were same

Macbooks-MacBook:~ macbook$ brew update
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "git".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
Error: Git must be installed and in your PATH!

and xcode is installed
pls help

**Comment by carlocab** [comment, marked as answer]
What do `brew config` and `brew doctor` say?

**Reply by m1sfumo** [reply]
`Macbooks-MacBook:~ macbook$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libspeedhack.dylib

Warning: You are using macOS 10.11.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

Error: No available formula with the name "git".`
brew doctor

**Reply by carlocab** [reply]
And `brew config`?

**Reply by gromgit** [reply]
Also run `brew readall`. It's looking more and more like one or more formulae in your local taps is b0rken.

**Reply by m1sfumo** [reply]
Macbooks-MacBook:~ macbook$ brew config
HOMEBREW_VERSION: 2.7.5-74-g7242cd6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7242cd64743635d13d5f69fa3ae0718b2170e7da
Last commit: 7 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.k70eNoCPHM/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 2
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dual-core 64-bit penryn
Clang: 8.0 build 800
Git: 2.10.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.43.0 => /usr/bin/curl
Java: 13.0.2, 1.6.0_65-b14-468
macOS: 10.11.6-x86_64
CLT: 8.2.0.0.1.1480973914
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11


brew readall no response

**Reply by carlocab** [reply]
```
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
```
Seems like you have no local core tap? It looks like there is no "git" formula because you have no formulae at all. Try
```
rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
brew tap homebrew/core
```

**Reply by m1sfumo** [reply]
thx kind of work but spammed with warning i'll make this answered
```
Macbooks-MacBook:~ macbook$ rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
Macbooks-MacBook:~ macbook$ brew tap homebrew/core
Error: Git must be installed and in your PATH!
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 27, done.
remote: Counting objects: 100% (27/27), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 890659 (delta 14), reused 16 (delta 8), pack-reused 890632
Receiving objects: 100% (890659/890659), 347.56 MiB | 1.52 MiB/s, done.
Resolving deltas: 100% (603910/603910), done.
Checking out files: 100% (5743/5743), done.
Tapped 2 commands and 5454 formulae (5,766 files, 382MB).
Macbooks-MacBook:~ macbook$ brew update
Warning: You are using macOS 10.11.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

Error: git: no bottle available!
You can try to install from source with:
  **brew install --build-from-source git**
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
Error: Git must be installed and in your PATH!
```

**Reply by hetansh1** [reply]
> ```
> Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
> Core tap HEAD: (none)
> Core tap last commit: never
> Core tap branch: HEAD
> ```
> 
> Seems like you have no local core tap? It looks like there is no "git" formula because you have no formulae at all. Try
> 
> ```
> rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
> brew tap homebrew/core
> ```

brew tap homebrew/core  ....returns me below output please help

Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin` exited with 128.

**Reply by gromgit** [reply]
> fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number

That's new. What's the output of:
```
git config -l
git clone https://github.com/gromgit/homebrew-dev /tmp/test-repo
```

**Reply by hetansh1** [reply]
> > fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number
> 
> That's new. What's the output of:
> 
> ```
> git config -l
> git clone https://github.com/gromgit/homebrew-dev /tmp/test-repo
> ```

Solved the issue -- removed brew and all the files from directories and installed it again... 
--and It Works like a charm now

**Reply by hetansh1** [reply]
> Hi,
> 
> I am facing the same error.
> 
> `brew config` returns
> 
> ```
> HOMEBREW_VERSION: 3.1.5-2-g302087a
> ORIGIN: https://github.com/Homebrew/brew
> HEAD: 302087aa1e784c4b6eda48fb94022d364824bf51
> Last commit: 9 hours ago
> Core tap ORIGIN: https://github.com/Homebrew/brew
> Core tap HEAD: 302087aa1e784c4b6eda48fb94022d364824bf51
> Core tap last commit: 9 hours ago
> Core tap branch: master
> HOMEBREW_PREFIX: /usr/local
> HOMEBREW_CASK_OPTS: []
> HOMEBREW_MAKE_JOBS: 12
> Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
> CPU: dodeca-core 64-bit kabylake
> Clang: 12.0 build 1205
> Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
> Curl: 7.64.1 => /usr/bin/curl
> macOS: 11.3-x86_64
> CLT: 12.5.0.0.1.1617976050
> Xcode: N/A
> ```
> 
> `brew doctor`
> 
> ```
> Please note that these warnings are just used to help the Homebrew maintainers
> with debugging if you file an issue. If everything you use Homebrew for is
> working fine: please don't worry or file an issue; just ignore this. Thanks!
> 
> Warning: Suspicious https://github.com/Homebrew/homebrew-core git origin remote found.
> The current git origin is:
>   https://github.com/Homebrew/brew
> 
> With a non-standard origin, Homebrew won't update properly.
> You can solve this by setting the origin remote:
>   git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin https://github.com/Homebrew/homebrew-core
> ```
> 
> tried `git -C "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin` then when `brew doctor` again it ask me to change back to the previous origin remote.
> 
> `brew readall` doesn't return anything
> 
> Tried removing brew and installing it a couple of times but still having the error "No available formulae"
> 
> Would appreciate any help here.

I solved issue by removing all the files in brew directory and reinstalling again (worked after 10th try though)
try after doing git confing (username and email) globaly - that might solve your problem
Good Luck <3

**Reply by kyeap** [reply]
I solved mine with 

`rm -rf /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core; brew update`

https://github.com/Homebrew/brew/issues/2387#issuecomment-288815977

**Reply by Ogwara** [reply]
Mine resolved. Thanks
